front/tests/pages/operational-studies-page-model.ts

```diff
@@ -44,9 +173,203 @@ class OperationalStudiesPage {
     await this.getEmptyRollingStockSelector.click();
   }
 
-  // Clicks to open the rolling stock selector
+  // Clicks to open the rolling stock selector.
   async openRollingStockSelector() {
     await this.getRollingStockSelector.click();
   }
+
+  // Clicks the button to close the via modal.
+  async clickOnCloseViaModalButton() {
+    await this.closeViaModalButton.click();
+  }
+
+  // Clicks the button to submit the search by trigram.
+  async clickSearchByTrigramSubmitButton() {
+    await this.searchByTrigramSubmit.click();
+  }
+
+  // Clicks the button to delete the itinerary.
+  async clickDeleteItineraryButton() {
+    await this.deleteItineraryButton.click();
+  }
+
+  // Verifies that no route is selected and displays appropriate messages based on language.
+  async verifyNoSelectedRoute(selectedLanguage: string) {
+    const noOriginChosenText = await this.noOriginChosen.innerText();
+    const noDestinationChosenText = await this.noDestinationChosen.innerText();
+    const isSearchByTrigramInputVisible = await this.searchByTrigramInput.isVisible();
+
+    const translations = selectedLanguage === 'English' ? enTranslations : frTranslations;
+
+    if (!isSearchByTrigramInputVisible) {
+      const pathfindingNoStateText = await this.pathfindingNoState.innerText();
+      expect(pathfindingNoStateText).toEqual(translations.pathfindingNoState);
+    }
+
+    expect(noOriginChosenText).toEqual(translations.noOriginChosen);
+    expect(noDestinationChosenText).toEqual(translations.noDestinationChosen);
+  }
+
+  // Performs pathfinding by entering origin, destination, and optionally via trigrams.
+  async performPathfindingByTrigram(
+    originTrigram: string,
+    destinationTrigram: string,
+    viaTrigram?: string
+  ) {
+    await this.searchByTrigramButton.click();
+    await expect(this.searchByTrigramContainer).toBeVisible();
+
+    const inputTrigramText = viaTrigram
+      ? `${originTrigram} ${viaTrigram} ${destinationTrigram}`
+      : `${originTrigram} ${destinationTrigram}`;
+    await this.searchByTrigramInput.fill(inputTrigramText);
+
+    if (viaTrigram) {
+      await expect(this.getOriginLocatorByTrigram(originTrigram)).toBeVisible();
+      await expect(this.getViaLocatorByTrigram(viaTrigram)).toBeVisible();
+      await expect(this.getDestinationLocatorByTrigram(destinationTrigram)).toBeVisible();
+    } else {
+      await expect(this.getOriginLocatorByTrigram(originTrigram)).toBeVisible();
+      await expect(this.getDestinationLocatorByTrigram(destinationTrigram)).toBeVisible();
+    }
```

Comment:
I think you can change like this:
```suggestion
    await expect(this.getOriginLocatorByTrigram(originTrigram)).toBeVisible();
    await expect(this.getDestinationLocatorByTrigram(destinationTrigram)).toBeVisible();
    if (viaTrigram) {
      await expect(this.getViaLocatorByTrigram(viaTrigram)).toBeVisible();
    }
```